### PR TITLE
Initial commit of optional hooks + manage_hook script

### PR DESCRIPTION
**Pull request by Spiker985**
Two hooks are available at the moment: post-checkout and post-merge.

- post-checkout fires when the branch changes, or when you checkout a new branch
- post-merge fires when you merge a branch into your existing branch

HOW_TO_USE.md contains some generic information on how to use `manage_hooks.sh`

`manage_hooks.sh` needs to be tested on MacOS, and also needs to be tested more thoroughly on other Linux builds

`update_editor.sh` is the actual brain. the post-* files are simply wrappers around update_editor.sh because of [Don't Repeat Yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)

### CODE REVIEW

File: hooks/manage_hooks-completion.bash [hunk 1]
```diff
@@ -0,0 +1,59 @@
+#!/bin/bash
+
+#Not gonna lie, this is some black magic, but I will try to explain it
```
Comment (mauricioszabo) [inline hooks/manage_hooks-completion.bash:3]:
```suggestion
#Not gonna lie, this is some magical stuff, but I will try to explain it
```

**Review by mauricioszabo — CHANGES_REQUESTED** [review]
I don't care much about hooks, so I'm assuming you tested them and they work.

I only suggest changing a word on the comments.

**Review by DeeDeeG — COMMENTED** [review]
I might suggest changing HOW_TO_USE.md to be README.md so GitHub users will get a nice rendered Markdown if they browse there.

Also suggest changing the "hardlink" terminology to be "copy", since I think a hardlink is something else -- I think it means the actual file data exists once on disk but multiple paths/filenames can bring up the same file's data.

https://en.wikipedia.org/wiki/Hard_link

We might want to put a disclaimer that an effort was made to put these together for convenience, but results are not guaranteed, and you run them at your own risk. Just on the safe side, since people might trust the repo's content as far as they trust the project overall, and this isn't a super supported or tested thing at the moment.

I am neutral on including it, but if an additional person wants this, I would then see more benefit than harm in committing it.

**Comment by Spiker985** [comment]
> Also suggest changing the "hardlink" terminology to be "copy", since I think a hardlink is something else -- I think it means the actual file data exists once on disk but multiple paths/filenames can bring up the same file's data. (All the hardlinks point to the same file data directly, without traversing shortcuts like with a softlink.)

Ironically, I didn't actually realize that. What's the point of symbolic vs hard linking if both of them technically just intrinsically link the same object together?

~~Though I guess a symbolic link is a redirection, and a hard link is a block:block link~~

**Comment by Spiker985** [comment]
Renamed markdown. Updated terminology. Replaced hardlink with copy

**Review by mauricioszabo — APPROVED** [review]

**Comment by mauricioszabo** [comment]
>  What's the point of symbolic vs hard linking if both of them technically just intrinsically link the same object together?

@Spiker985 a hard link is indistinguishable from a real file - that's why it's a "hard" link. So, if for example, you have a 1gb file called `myfile.dat` and you hardlink it as `othername.dat`, you'll occupy only 1gb on disk and if you delete only one of them, the other will still work and you will still have 1gb of space occupied on your link.

A symbolic link can be detected by programs (and by the OS) and only the "original" file is the real one - of you delete the "original" file you will invalidate all other links.

There are other differences, but basically it's kinda like a symlink is a "path" to a real file, where a hard link is _a real file_ but sharing the space of other.

**Comment by Spiker985** [comment]
Since this only adds new optional content, Mauricio approved the adjustments, and I implemented Dee's suggestions, I'm going to merge it

**Pull request merged**